public/app/features/plugins/components/PluginErrorBoundary.test.tsx

```diff
@@ -0,0 +1,173 @@
+import { render, screen } from '@testing-library/react';
+import * as React from 'react';
+
+import { PluginMeta, PluginType, PluginContext } from '@grafana/data';
+import { getMockPlugin } from '@grafana/data/test';
+
+import { PluginErrorBoundary } from './PluginErrorBoundary';
+
+const ThrowingComponent = ({ shouldThrow }: { shouldThrow: boolean }) => {
+  if (shouldThrow) {
+    throw new Error('Test error message');
+  }
+  return <div>Working component</div>;
+};
+
+const TestFallback = ({ error, errorInfo }: { error: Error | null; errorInfo: React.ErrorInfo | null }) => (
+  <div>
+    <div>Fallback rendered</div>
+    <div>Error: {error?.message}</div>
+    {errorInfo && <div>Error info available</div>}
+  </div>
+);
+
+const renderWithPluginContext = (
+  children: React.ReactNode,
+  pluginMeta?: PluginMeta,
+  fallback?: React.ComponentType<{ error: Error | null; errorInfo: React.ErrorInfo | null }>,
+  onError?: (error: Error, info: React.ErrorInfo) => void
+) => {
```

Comment:
[nitpick]: could we use an object with these props instead, hard to read and follow the code?
```suggestion
const renderWithPluginContext = ({
children,
pluginMeta = getMockPlugin({ id: 'test-plugin', type: PluginType.panel }),
fallback,
onError
}: {
  children: React.ReactNode;
  pluginMeta?: PluginMeta;
  fallback?: React.ComponentType<{ error: Error | null; errorInfo: React.ErrorInfo | null }>;
  onError?: (error: Error, info: React.ErrorInfo) => void;
}) => {
```